### PR TITLE
improve "insert_estimation_task"; loading DLC results handles multiple DLC output files

### PR DESCRIPTION
**Pull request by ttngu207**
+ Fix - improve function to auto generate PoseEstimationTask
+ Update - loading DLC results handles multiple DLC output files

### CODE REVIEW

File: element_deeplabcut/readers/dlc_reader.py [hunk 1]
```diff
@@ -33,33 +33,38 @@ def __init__(
...
         if h5_path is None:
-            h5_paths = list(self.dlc_dir.rglob(f"{filename_prefix}*.h5"))
-            assert len(h5_paths) == 1, (
-                "Unable to find one unique .h5 file in: "
-                + f"{dlc_dir} - Found: {len(h5_paths)}"
-            )
-            self.h5_path = h5_paths[0]
+            self.h5_paths = sorted(self.dlc_dir.rglob(f"{filename_prefix}*.h5"))
+            assert (
+                len(self.h5_paths) > 1
```
Comment (CBroz1) [inline element_deeplabcut/readers/dlc_reader.py:53]: Do you want more than one here?
```suggestion
                len(self.h5_paths) > 0
```

File: element_deeplabcut/model.py [hunk 2]
```diff
@@ -603,41 +602,53 @@ def infer_output_dir(cls, key: dict, relative: bool = False, mkdir: bool = False
...
-        output_dir = cls.infer_output_dir(key, relative=relative, mkdir=mkdir)
+        processed_dir = get_dlc_processed_data_dir()
+        output_dir = cls.infer_output_dir(
+            video_recording_key, relative=False, mkdir=True
+        )
+
+        if task_mode is None:
+            try:
+                _ = dlc_reader.PoseEstimation(output_dir)
+            except FileNotFoundError:
```
Comment (CBroz1) [inline element_deeplabcut/model.py:633]: Are you sure this would hit `FileNotFound`? - I think we use assertions to ensure requisite files are available.
```suggestion
            except AssertionError:

```

File: element_deeplabcut/model.py [hunk 3]
```diff
@@ -683,28 +693,41 @@ def make(self, key):
         # ID model and directories
         dlc_model = (Model & key).fetch1()
 
-        task_mode, analyze_video_params, output_dir = (PoseEstimationTask & key).fetch1(
-            "task_mode", "pose_estimation_params", "pose_estimation_output_dir"
+        assert dlc_model["project_path"], (
+            "Your model table must have a 'project_path'"
+            + "field pointing to a DLC directory"
```
Comment (CBroz1) [inline element_deeplabcut/model.py:698]: Why is this necessary if the [table field](https://github.com/ttngu207/element-deeplabcut/blob/main/element_deeplabcut/model.py#L332) is not nullable? Should we instead be asserting here that the path exists?

File: element_deeplabcut/model.py [hunk 4]
```diff
@@ -683,28 +693,41 @@ def make(self, key):
...
+            project_path = find_full_path(
+                get_dlc_root_data_dir(), dlc_model["project_path"]
+            )
+            video_filepaths = [
+                find_full_path(get_dlc_root_data_dir(), fp).as_posix()
+                for fp in (VideoRecording.File & key).fetch("file_path")
+            ]
+            analyze_video_params = (PoseEstimationTask & key).fetch1(
+                "pose_estimation_params"
+            ) or {}
```
Comment (CBroz1) [inline element_deeplabcut/model.py:721]: I think this `fetch1` would throw an error if it didn't get exactly one entry
```suggestion
            analyze_video_params = (PoseEstimationTask & key).fetch(
                "pose_estimation_params", limit=1
            )[0] or {}

```

**Review by CBroz1 — CHANGES_REQUESTED** [review]
Thanks for these additions, @ttngu207. Looks good in general, but I had some questions about error handling

File: element_deeplabcut/readers/dlc_reader.py (hunk 1 above)
Comment (ttngu207) [reply element_deeplabcut/readers/dlc_reader.py:53]: `> 0` is what I meant, thanks for the catch

File: element_deeplabcut/model.py (hunk 2 above)
Comment (ttngu207) [reply element_deeplabcut/model.py:633]: Right, I'll change those to `FileNotFound` for better error clarity

File: element_deeplabcut/model.py (hunk 3 above)
Comment (ttngu207) [reply element_deeplabcut/model.py:698]: You're right, we don't need this

File: element_deeplabcut/model.py (hunk 4 above)
Comment (ttngu207) [reply element_deeplabcut/model.py:721]: How so? There must be only one unique entry in `PoseEstimationTask` associated with this `key`

File: element_deeplabcut/readers/dlc_reader.py (hunk 1 above)
Comment (ttngu207) [reply element_deeplabcut/readers/dlc_reader.py]: fixed

File: element_deeplabcut/model.py (hunk 2 above)
Comment (ttngu207) [reply element_deeplabcut/model.py:634]: fixed

File: element_deeplabcut/model.py (hunk 3 above)
Comment (ttngu207) [reply element_deeplabcut/model.py]: fixed

File: element_deeplabcut/model.py (hunk 4 above)
Comment (CBroz1) [reply element_deeplabcut/model.py:717]: My mistake - The `fetch1` here would return the `null` stored under `pose_estimation_params`

**Review by CBroz1 — APPROVED** [review]

**Comment by CBroz1** [comment]
@ttngu207 - Do you want to run pytests before merging?

**Review by sidhulyalkar — APPROVED** [review]
